src/openforms/js/components/admin/form_design/registrations/objectsapi/AddressNlObjectsApiVariableConfigurationEditor.js

```diff
@@ -0,0 +1,347 @@
+            >
+              <TargetPathSelect
+                id="cityTargetPath"
+                name={`${namePrefix}.cityTargetPath`}
+                index={index}
+                choices={choices}
+                mappedVariable={mappedVariable}
+                disabled={!deriveAddress}
+              />
+            </Field>
+          </FormRow>
+          <FormRow>
+            <Field
+              name={`${namePrefix}.streetNameTargetPath`}
+              label={
+                <FormattedMessage
+                  defaultMessage="Street name Schema target"
+                  description="'Street name Schema target' label"
+                />
+              }
+              disabled={!deriveAddress}
+            >
+              <TargetPathSelect
+                id="streetNameTargetPath"
+                name={`${namePrefix}.streetNameTargetPath`}
+                index={index}
+                choices={choices}
+                mappedVariable={mappedVariable}
+                disabled={!deriveAddress}
+              />
+            </Field>
+          </FormRow>
+        </Fieldset>
+      )}
+      <div style={{marginTop: '1em'}}>
+        <a href="#" onClick={e => e.preventDefault() || toggleJsonSchemaVisible()}>
+          <FormattedMessage
+            description="Objects API variable configuration editor JSON Schema visibility toggle"
+            defaultMessage="Toggle JSON Schema"
+          />
+        </a>
+        {jsonSchemaVisible && (
+          <pre style={{marginTop: '1em'}}>
+            {loading || (!mappedVariable.targetPath && !isSpecificTargetPaths) ? (
+              <FormattedMessage description="Not applicable" defaultMessage="N/A" />
+            ) : isSpecificTargetPaths ? (
+              JSON.stringify(ADDRESSNL_NESTED_PROPERTIES, null, 2)
+            ) : (
+              JSON.stringify(getTargetPath(mappedVariable.targetPath).jsonSchema, null, 2)
+            )}
+          </pre>
+        )}
+      </div>
+    </>
+  );
+};
+
+const TargetPathSelect = ({id, name, index, choices, mappedVariable, disabled}) => {
+  // To avoid having an incomplete variable mapping added in the `variablesMapping` array,
+  // It is added only when an actual target path is selected. This way, having the empty
+  // option selected means the variable is unmapped (hence the `arrayHelpers.remove` call below).
+  const {
+    values: {variablesMapping},
+    getFieldProps,
+    setFieldValue,
+  } = useFormikContext();
+  const props = getFieldProps(name);
+  const isNew = variablesMapping.length === index;
+
+  return (
+    <FieldArray
+      name="variablesMapping"
+      render={arrayHelpers => (
+        <Select
+          id={id}
+          name={name}
+          allowBlank
+          choices={choices}
+          {...props}
+          disabled={disabled}
+          value={JSON.stringify(props.value)}
+          onChange={event => {
+            if (event.target.value === '') {
+              arrayHelpers.remove(name);
```

Comment:
Same as https://github.com/open-formulieren/open-forms/pull/4751/files#r1832285232. The entire dataset will be removed if one value is changed back to empty